transcom/mymove
# B-20550

## [20550](https://www13.v1host.com/USTRANSCOM38/story.mvc/Summary?oidToken=Story%3A981778)

## Summary

The date in question is in the GS segment of the 858. Currently that date is set to the currentTime at the time of invoice generation
- Updated the current date and current time segment of GS to the requested Date, time
## Verification Steps for the Author

These are to be checked by the author.


## Verification Steps for Reviewers

These are to be checked by a reviewer.


### Setup to Run the Code

- [Instructions for starting storybook](https://transcom.github.io/mymove-docs/docs/frontend/setup/storybook)
- [Instructions for starting the MilMove application](https://transcom.github.io/mymove-docs/docs/getting-started/application-setup/)
- [Instructions for running tests](https://transcom.github.io/mymove-docs/docs/getting-started/development/testing)

### How to test

1. Access the milmovelocal - testharness scenarios
2. select MakeMoveReadyForEDI
3. 
![image](https://github.com/user-attachments/assets/a510a255-c40c-4d73-9b4c-456ccdc58340)

4. grab the payment_request_number
5. 
![image](https://github.com/user-attachments/assets/0c9694cf-544a-4cc7-9fa3-7033080d0e84)

6. open terminal and run `go run ./cmd/generate-payment-request-edi/ --payment-request-number <payment_request_number>`
7. compare the requested_at field  on the json to the GS field for date and time to see they match
00010101*0000

![image](https://github.com/user-attachments/assets/cc86dd89-c552-450d-a5a3-72cbdda3b45b)

Optional- update the requested_at field in DB (payment_request) to see a more realistic date
![image](https://github.com/user-attachments/assets/bc314ee1-ea9b-4aa1-973b-66676cded261)

![image](https://github.com/user-attachments/assets/036c3972-b56c-4873-a095-ff0623da5ea9)



### Backend
